Feature/e2e

saulipurhonen opened the pull request:
### Description

Basic e2e test that covers:
- Folder creation
- Form save
- XML upload
- Wizard navigation
- Folder publishing

To run, execute **npx cypress open**

### Related issues

Closes #10 

### Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update (this needs a follow up PR)

### Changes Made

- Installed Cypress and cypress-file-upload packages
- Tested frontend logic with backend connection

### Testing

- [x] Tests do not apply

saulipurhonen commented:
Note: There isn't much more to test in current state of the app. This merge can be used as base for e2e tests with Cypress.

Also, CI and GH Actions integration aren't in this PR. Should we try to get those running as well?

blankdots commented:
> Also, CI and GH Actions integration aren't in this PR. Should we try to get those running as well?

yes I think those should be done as well this could be introduced as part of this PR or another PR  (if that is the case please create an issue about what will be done)
https://github.com/cypress-io/github-action - seems to be the way to go ?

Note: the workflow will change following https://github.com/CSCfi/metadata-submitter/pull/158 and https://github.com/CSCfi/metadata-submitter/pull/151

as the user would need login, but a mock AAI is provided in the integration tests of those PRs.

blankdots commented on cypress/integration/app.spec.js:
```diff
@@ -0,0 +1,46 @@
+    cy.get("select[name='descriptor.studyType']").select("Metagenomics")
+    cy.get("button[type=submit]").contains("Save").click()
+
+    // Upload an xml file. Need to wait for save. TODO: Better logic for waiting until last save is done
```
please do an issue detailing this To Do.

blankdots commented on CONTRIBUTING.md:
```diff
@@ -6,6 +6,8 @@ If you're reading this, it means you saw something that is not right, you want t
 
 We have a handful of unit tests and integration tests. In our Git workflow unit tests are run on every Push and Pull Request, while integration tests run on every Pull Request.
 
+We use Cypress for e2e testing.
```
```suggestion
We use [Cypress](https://www.cypress.io/) for e2e testing.
```

blankdots reviewed (changes requested):
I am ok with leaving `This example ...` in the cypress files but I must recognise don't look very nice.

We would reference the source and where to find more information.

the only requests of change would be:
- to have a github action for this.
- mention in main `README.md` how to run the tests

saulipurhonen commented:
> I am ok with leaving `This example ...` in the cypress files but I must recognise don't look very nice.
> 
> We would reference the source and where to find more information.
> 
> the only requests of change would be:
> 
> * to have a github action for this.
> * mention in main `README.md` how to run the tests

I agree that example file doesn't look good.
I Removed the file and added e2e testing info to README.md

csc-jm reviewed (approved):
LGTM
